### PR TITLE
feat(sinsp/threadinfo): expose thread uid and gid as static fields

### PR DESCRIPTION
**Pull request by ekoops**
**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

> /kind bug

> /kind cleanup

> /kind design

> /kind documentation

> /kind failing-test

/kind feature

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

> /area API-version

> /area build

> /area CI

> /area driver-kmod

> /area driver-bpf

> /area driver-modern-bpf

> /area libscap-engine-bpf

> /area libscap-engine-gvisor

> /area libscap-engine-kmod

> /area libscap-engine-modern-bpf

> /area libscap-engine-nodriver

> /area libscap-engine-noop

> /area libscap-engine-source-plugin

> /area libscap-engine-savefile

> /area libscap

> /area libpman

/area libsinsp

> /area tests

> /area proposals

**Does this PR require a change in the driver versions?**

> /version driver-API-version-major

> /version driver-API-version-minor

> /version driver-API-version-patch

> /version driver-SCHEMA-version-major

> /version driver-SCHEMA-version-minor

> /version driver-SCHEMA-version-patch

**What this PR does / why we need it**:
This PR enables access to thread `uid` and `gid` as static fields.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: userspace/libsinsp/threadinfo.cpp [hunk 1]
```diff
@@ -77,6 +77,8 @@ libsinsp::state::static_struct::field_infos sinsp_threadinfo::static_fields() co
 	define_static_field(ret, this, m_container_id, "container_id");
 	define_static_field(ret, this, m_flags, "flags");
 	define_static_field(ret, this, m_fdlimit, "fd_limit");
+	define_static_field(ret, this, m_uid, "uid");
+	define_static_field(ret, this, m_gid, "gid");
```
Comment (FedeDP) [inline userspace/libsinsp/threadinfo.cpp:81]: `login_uid` too?

File: userspace/libsinsp/threadinfo.cpp [hunk 2]
```diff
@@ -77,6 +77,8 @@ libsinsp::state::static_struct::field_infos sinsp_threadinfo::static_fields() co
...
 	// m_user
```
Comment (FedeDP) [inline userspace/libsinsp/threadinfo.cpp:82]: Also let's cleanup these 3 lines below.

**Review by FedeDP — DISMISSED** [review]
/approve

**Review by Andreagit97 — DISMISSED** [review]
/approve

**Comment by Andreagit97** [comment]
You need to update the corresponding test
```
2024-12-10T16:27:02.9330936Z [ RUN      ] thread_manager.table_access
2024-12-10T16:27:02.9331451Z /__w/libs/libs/userspace/libsinsp/test/state.ut.cpp:381: Failure
2024-12-10T16:27:02.9331926Z Expected equality of these values:
2024-12-10T16:27:02.9332533Z   newt->static_fields().size()
2024-12-10T16:27:02.9332834Z     Which is: 33
2024-12-10T16:27:02.9333103Z   s_threadinfo_static_fields_count
2024-12-10T16:27:02.9333628Z     Which is: 30
2024-12-10T16:27:02.9333924Z [  FAILED  ] thread_manager.table_access (1 ms)
2024-12-10T16:27:02.9334319Z [ RUN      ] thread_manager.fdtable_access
```

**Comment by ekoops** [comment]
> You need to update the corresponding test
> 
> ```
> 2024-12-10T16:27:02.9330936Z [ RUN      ] thread_manager.table_access
> 2024-12-10T16:27:02.9331451Z /__w/libs/libs/userspace/libsinsp/test/state.ut.cpp:381: Failure
> 2024-12-10T16:27:02.9331926Z Expected equality of these values:
> 2024-12-10T16:27:02.9332533Z   newt->static_fields().size()
> 2024-12-10T16:27:02.9332834Z     Which is: 33
> 2024-12-10T16:27:02.9333103Z   s_threadinfo_static_fields_count
> 2024-12-10T16:27:02.9333628Z     Which is: 30
> 2024-12-10T16:27:02.9333924Z [  FAILED  ] thread_manager.table_access (1 ms)
> 2024-12-10T16:27:02.9334319Z [ RUN      ] thread_manager.fdtable_access
> ```

Done

**Review by FedeDP — APPROVED** [review]
/approve

**Review by Andreagit97 — APPROVED** [review]
/approve

**Pull request merged**